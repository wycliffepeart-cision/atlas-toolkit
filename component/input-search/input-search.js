import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var InputSearchTemplate = "<template>\n  <input id=\"Input\" type=\"text\" part=\"input\">\n\n  <label for=\"Input\" part=\"label\"><slot name=\"label\"></slot></label>\n\n  <slot name=\"prefix\" part=\"prefix\"></slot>\n  <slot name=\"suffix\" part=\"suffix\"></slot>\n\n  <slot name=\"menu\" part=\"menu\"></slot>\n</template>\n";

const SEARCH_ATTRIBUTES = [
    'disabled',
    'maxlength',
    'pattern',
    'placeholder',
    'required',
    'readonly',
    'size',
    'value',
    'incremental',
];
const SEARCH_EVENTS = ['blur', 'focus', 'keydown', 'keyup', 'input', 'change'];
let InputSearch = class InputSearch extends HTMLElement {
    static get observedAttributes() {
        return [...SEARCH_ATTRIBUTES];
    }
    connectedCallback() {
        if (!this.isConnected) {
            return;
        }
        const input = this.shadowRoot.getElementById('Input');
        input.addEventListener('change', this.#handleOnChange);
    }
    disconnectedCallback() {
        const input = this.shadowRoot.getElementById('Input');
        input.removeEventListener('change', this.#handleOnChange);
    }
    #handleOnChange = () => {
        this.dispatchEvent(new CustomEvent('change', { bubbles: true, composed: true }));
    };
    attributeChangedCallback(name, _oldValue, newValue) {
        if (SEARCH_ATTRIBUTES.includes(name.trim().toLowerCase())) {
            const input = this.shadowRoot.getElementById('Input');
            if (newValue === null) {
                input.removeAttribute(name);
            }
            else {
                input.setAttribute(name, newValue);
            }
        }
        // unknown attribute?
    }
    set value(text) {
        const input = this.shadowRoot.getElementById('Input');
        input.value = text;
    }
    get value() {
        const input = this.shadowRoot.getElementById('Input');
        return input.value;
    }
    set disabled(value) {
        // for toggle attributes, use toggle
        this.toggleAttribute('disabled', value);
    }
    get disabled() {
        // because disabled is a toggle attribute, its presents is all that matters
        return this.hasAttribute('disabled');
    }
};
InputSearch = __decorate([
    Template(InputSearchTemplate)
], InputSearch);
if (!customElements.get('atlas-search'))
    customElements.define('atlas-search', InputSearch);

export { InputSearch, SEARCH_ATTRIBUTES, SEARCH_EVENTS };
