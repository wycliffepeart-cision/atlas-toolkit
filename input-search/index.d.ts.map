{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../src/input-search/index.ts"],"names":[],"mappings":""}