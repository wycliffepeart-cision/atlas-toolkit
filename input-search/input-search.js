import { _ as __decorate, a as __classPrivateFieldGet } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var InputSearchTemplate = "<template>\n  <input id=\"Input\" type=\"text\" part=\"input\">\n\n  <label for=\"Input\" part=\"label\"><slot name=\"label\"></slot></label>\n\n  <slot name=\"prefix\" part=\"prefix\"></slot>\n  <slot name=\"suffix\" part=\"suffix\"></slot>\n\n  <slot name=\"menu\" part=\"menu\"></slot>\n</template>\n";

var _InputSearch_handleOnChange;
const SEARCH_ATTRIBUTES = [
    'disabled',
    'maxlength',
    'pattern',
    'placeholder',
    'required',
    'readonly',
    'size',
    'value',
    'incremental',
];
const SEARCH_EVENTS = ['blur', 'focus', 'keydown', 'keyup', 'input', 'change'];
let InputSearch = class InputSearch extends HTMLElement {
    constructor() {
        super(...arguments);
        _InputSearch_handleOnChange.set(this, () => {
            this.dispatchEvent(new CustomEvent('change', { bubbles: true, composed: true }));
        });
    }
    static get observedAttributes() {
        return [...SEARCH_ATTRIBUTES];
    }
    connectedCallback() {
        if (!this.isConnected) {
            return;
        }
        const input = this.shadowRoot.getElementById('Input');
        input.addEventListener('change', __classPrivateFieldGet(this, _InputSearch_handleOnChange, "f"));
    }
    disconnectedCallback() {
        const input = this.shadowRoot.getElementById('Input');
        input.removeEventListener('change', __classPrivateFieldGet(this, _InputSearch_handleOnChange, "f"));
    }
    attributeChangedCallback(name, _oldValue, newValue) {
        if (SEARCH_ATTRIBUTES.includes(name.trim().toLowerCase())) {
            const input = this.shadowRoot.getElementById('Input');
            if (newValue === null) {
                input.removeAttribute(name);
            }
            else {
                input.setAttribute(name, newValue);
            }
        }
        // unknown attribute?
    }
    set value(text) {
        const input = this.shadowRoot.getElementById('Input');
        input.value = text;
    }
    get value() {
        const input = this.shadowRoot.getElementById('Input');
        return input.value;
    }
    set disabled(value) {
        // for toggle attributes, use toggle
        this.toggleAttribute('disabled', value);
    }
    get disabled() {
        // because disabled is a toggle attribute, its presents is all that matters
        return this.hasAttribute('disabled');
    }
};
_InputSearch_handleOnChange = new WeakMap();
InputSearch = __decorate([
    Template(InputSearchTemplate)
], InputSearch);

export { InputSearch, SEARCH_ATTRIBUTES, SEARCH_EVENTS };
