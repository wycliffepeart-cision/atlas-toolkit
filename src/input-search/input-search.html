<template>
	<input id="Input" type="text" part="input" />

	<label for="Input" part="label"><slot name="label"></slot></label>

	<slot name="prefix" part="prefix"></slot>
	<slot name="suffix" part="suffix"></slot>

	<slot name="menu" part="menu"></slot>
</template>
