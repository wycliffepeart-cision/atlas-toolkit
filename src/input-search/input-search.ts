import { Template } from '../decorators/template';
import InputSearchTemplate from './input-search.html';

export const SEARCH_ATTRIBUTES = [
  'disabled',
  'maxlength',
  'pattern',
  'placeholder',
  'required',
  'readonly',
  'size',

  'value',

  'incremental',
];

export const SEARCH_EVENTS = ['blur', 'focus', 'keydown', 'keyup', 'input', 'change'];

@Template(InputSearchTemplate)
export class InputSearch extends HTMLElement {
  static get observedAttributes() {
    return [...SEARCH_ATTRIBUTES];
  }

  connectedCallback() {
    if (!this.isConnected) {
      return;
    }
    const input = this.shadowRoot.getElementById('Input');
    input.addEventListener('change', this.#handleOnChange);
  }

  disconnectedCallback() {
    const input = this.shadowRoot.getElementById('Input');
    input.removeEventListener('change', this.#handleOnChange);
  }

  #handleOnChange = () => {
    this.dispatchEvent(new CustomEvent('change', { bubbles: true, composed: true }));
  };

  attributeChangedCallback(name, _oldValue, newValue) {
    if (SEARCH_ATTRIBUTES.includes(name.trim().toLowerCase())) {
      const input = this.shadowRoot.getElementById('Input');
      if (newValue === null) {
        input.removeAttribute(name);
      } else {
        input.setAttribute(name, newValue);
      }
    }

    // unknown attribute?
  }

  set value(text) {
    const input = this.shadowRoot.getElementById('Input') as HTMLInputElement;
    input.value = text;
  }

  get value() {
    const input = this.shadowRoot.getElementById('Input') as HTMLInputElement;
    return input.value;
  }

  set disabled(value) {
    // for toggle attributes, use toggle
    this.toggleAttribute('disabled', value);
  }

  get disabled() {
    // because disabled is a toggle attribute, its presents is all that matters
    return this.hasAttribute('disabled');
  }
}
