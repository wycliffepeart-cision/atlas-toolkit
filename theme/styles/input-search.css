@layer atlas {
	atlas-search {
		--border-color: var(--atlas-color-primary-n100);

		--disabled-search-text-color: var(--atlas-color-primary-n400);
		--disabled-background-color: var(--atlas-color-primary-n50);

		--border-radius: 0.25em;
		--border: 1px solid var(--atlas-color-primary-n200);
		--search-text-color: var(--atlas-color-primary-n400);

		--search-text-font-weight: 400;
		--search-text-font-size: 1em;

		--overall-flow-space: 0.5em;
		--inner-space: 0.8em;

		--affix-color: var(--atlas-color-primary-n100);
		--affix-font-size: 0.75em;
		--affix-inline-margin: 1.25em;
		--affix-affordance: 3em;
	}

	atlas-search {
		display: grid;

		grid-template-columns: min-content 1fr min-content;
		grid-template-areas:
			"P I S"
			"M M M";

		margin-block: var(--overall-flow-space);
	}

	atlas-search::part(label) {
		display: none;
	}

	atlas-search::part(menu) {
		grid-area: M;
	}

	atlas-search:not([show-menu])::part(menu),
	atlas-search[show-menu="false"]::part(menu) {
		display: none;
	}

	atlas-search::part(prefix) {
		display: inline-block;
		grid-area: P;

		align-self: center;
		font-size: var(--affix-font-size);

		color: var(--affix-color);

		margin-inline-start: var(--affix-inline-margin);
	}

	atlas-search::part(suffix) {
		display: inline-block;
		grid-area: S;

		align-self: center;
		font-size: var(--affix-font-size);

		color: var(--affix-color);

		margin-inline-end: var(--affix-inline-margin);
	}

	atlas-search::part(input) {
		display: inline-block;

		grid-row: 1;
		grid-column: 1 / span 3;

		width: auto;
		margin: 0;
		padding-block: 0;

		font-weight: var(--search-text-font-weight);
		font-size: var(--search-text-font-size);

		color: var(--search-text-color);
		background-color: var(--atlas-input-common-background-color);
		border: var(--border);
		border-radius: var(--border-radius);

		padding-block: var(--inner-space);
		padding-inline: var(--affix-affordance);
	}

	atlas-search[disabled]:not([disabled="false"])::part(input) {
		background-color: var(--disabled-background-color);
		border-color: var(--disabled-background-color);
		color: var(--disabled-search-text-color);
		pointer-events: none;
		cursor: auto;
	}

	atlas-search[disabled]:not([disabled="false"])::part(suffix) {
		display: none;
	}

	atlas-search:focus-within::part(input) {
		box-shadow: var(--atlas-input-common-box-shadow);
		outline: none;
	}
}
