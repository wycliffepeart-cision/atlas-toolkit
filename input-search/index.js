import { InputSearch } from './input-search.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-search'))
    customElements.define('atlas-search', InputSearch);
