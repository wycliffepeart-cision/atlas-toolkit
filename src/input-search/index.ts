import { InputSearch } from './input-search';

if (!customElements.get('atlas-search')) customElements.define('atlas-search', InputSearch);
