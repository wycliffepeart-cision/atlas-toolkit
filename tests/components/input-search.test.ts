import '../../src/component/input-search/input-search';

const INPUT_SEARCH_TAG = 'atlas-search';

afterEach(() => {
  const elem = document.querySelector('body > *:first-child');
  elem?.remove();
});

describe('Input', () => {
  describe('Search JS API', () => {
    it('supports .value getter', async () => {
      await customElements.whenDefined(INPUT_SEARCH_TAG);
      const search: any = document.createElement(INPUT_SEARCH_TAG);

      (expect(search.value) as any).toEqual('');
    });

    it('supports .value setter', async () => {
      await customElements.whenDefined(INPUT_SEARCH_TAG);
      const search: any = document.createElement(INPUT_SEARCH_TAG);
      search.value = 'SOME FAKE STRING';

      (expect(search.value) as any).toEqual('SOME FAKE STRING');
    });

    it('supports .value from value attribute', async () => {
      await customElements.whenDefined(INPUT_SEARCH_TAG);
      const search: any = document.createElement(INPUT_SEARCH_TAG);
      search.setAttribute('value', 'SOME FAKE STRING');

      (expect(search.value) as any).toEqual('SOME FAKE STRING');
    });

    it('supports .disabled getter', async () => {
      await customElements.whenDefined(INPUT_SEARCH_TAG);
      const search: any = document.createElement(INPUT_SEARCH_TAG);

      (expect(search.disabled) as any).toEqual(false);
    });

    it('supports .disabled setter', async () => {
      await customElements.whenDefined(INPUT_SEARCH_TAG);
      const search: any = document.createElement(INPUT_SEARCH_TAG);
      search.disabled = true;

      (expect(search.disabled) as any).toEqual(true);
    });

    it('supports .disabled from attribute', async () => {
      await customElements.whenDefined(INPUT_SEARCH_TAG);
      const search: any = document.createElement(INPUT_SEARCH_TAG);
      search.toggleAttribute('disabled', true);

      (expect(search.disabled) as any).toEqual(true);
    });

    it('supports .disabled from non-toggle attribute', async () => {
      await customElements.whenDefined(INPUT_SEARCH_TAG);
      const search: any = document.createElement(INPUT_SEARCH_TAG);
      search.setAttribute('disabled', 'SOME_FAKE_VALUE');

      (expect(search.disabled) as any).toEqual(true);
    });

    it('supports .disabled from non-toggle attribute - false', async () => {
      await customElements.whenDefined(INPUT_SEARCH_TAG);
      const search: any = document.createElement(INPUT_SEARCH_TAG);
      search.setAttribute('disabled', 'false');

      (expect(search.disabled) as any).toEqual(true);
    });
  });

  describe('Search', () => {
    it('exists', async () => {
      await customElements.whenDefined(INPUT_SEARCH_TAG);
      const dashboardElement = document.createElement(INPUT_SEARCH_TAG);
      document.body.appendChild(dashboardElement);

      (expect(dashboardElement) as any).toBeInTheDocument();
    });

    it('is matches snapshot', () => {
      const elem = document.createElement(INPUT_SEARCH_TAG);
      document.body.appendChild(elem);

      (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
    });

    it('is matches snapshot when disabled', () => {
      const elem = document.createElement(INPUT_SEARCH_TAG);
      document.body.appendChild(elem);

      elem.toggleAttribute('disabled', true);

      (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
    });

    it('is matches snapshot when disabled toggled on/off', () => {
      const elem = document.createElement(INPUT_SEARCH_TAG);
      document.body.appendChild(elem);

      elem.toggleAttribute('disabled', true);
      elem.toggleAttribute('disabled', false);

      (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
    });
  });
});
