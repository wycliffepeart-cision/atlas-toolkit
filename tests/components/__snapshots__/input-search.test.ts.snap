// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Search is matches snapshot 1`] = `
<input id="Input"
       type="text"
       part="input"
>
<label for="Input"
       part="label"
>
  <slot name="label">
  </slot>
</label>
<slot name="prefix"
      part="prefix"
>
</slot>
<slot name="suffix"
      part="suffix"
>
</slot>
<slot name="menu"
      part="menu"
>
</slot>
`;

exports[`Input Search is matches snapshot when disabled 1`] = `
<input id="Input"
       type="text"
       part="input"
       disabled
>
<label for="Input"
       part="label"
>
  <slot name="label">
  </slot>
</label>
<slot name="prefix"
      part="prefix"
>
</slot>
<slot name="suffix"
      part="suffix"
>
</slot>
<slot name="menu"
      part="menu"
>
</slot>
`;

exports[`Input Search is matches snapshot when disabled toggled on/off 1`] = `
<input id="Input"
       type="text"
       part="input"
>
<label for="Input"
       part="label"
>
  <slot name="label">
  </slot>
</label>
<slot name="prefix"
      part="prefix"
>
</slot>
<slot name="suffix"
      part="suffix"
>
</slot>
<slot name="menu"
      part="menu"
>
</slot>
`;
